crates/settlement-clients/starknet/src/lib.rs

```diff
@@ -197,11 +206,26 @@ impl SettlementClient for StarknetSettlementClient {
         if block_number.is_empty() {
             return Err(eyre!("Could not fetch last block number from core contract."));
         }
-        Ok(block_number[0].try_into()?)
+
+        Ok(u64::from_le_bytes(block_number[0].to_bytes_le()[0..8].try_into().unwrap()))
```

Comment:
1. can you move this to a function in `conversions.rs`
2. the test case should call this function and the code as well
3. the function should throw an error if bytes from 8-32 are non zero because then the conversion will be invalid
4. nitpick: i have usually seen `be` being used more commonly and and it's easier to understand because that's how we write stuff as well, should we use that?

Reply:
1. Created a new function
2. Added a testcase for the conversion as well
3. the functions asserts if the value is within `u64` range and bytes from 8 to 31 are zero
4. using `be` now